module-gui/gui/core/DrawCommand.cpp

```diff
@@ -228,19 +228,26 @@ namespace gui
         uint32_t offsetRowContext = 0;
         uint32_t imageOffset      = 0;
         uint8_t alphaColor        = vecMap->getAlphaColor();
+        checkImageSize(ctx, vecMap);
 
         for (uint32_t row = 0; row < std::min(vecMap->getHeight(), ctx->getH()); row++) {
-            checkImageSize(ctx, vecMap);
+
             uint16_t vecCount = *(vecMap->getData() + imageOffset);
-            imageOffset += sizeof(uint16_t);
+            imageOffset += sizeof(uint8_t);
+            vecCount <<= 8;
+            vecCount |= *(vecMap->getData() + imageOffset);
+            imageOffset += sizeof(uint8_t);
```

Comment:
I've approved, as this code 'is as it is' though it might be a oneliner or lambda.
 Preferably complimented with a short test i.e. here: https://github.com/mudita/MuditaOS/tree/master/module-gui/test/test-catch

Please consider what you might think as "unnecessary prettifying" in this case :) Saing that - there is no reinterpret cast - therefore approved.

Reply:
A `constexpr` function that uses a `reinterpret_cast` with a target conditional byteswap seems like a better solution.  It optimizes to machine code much better too.  Byteswapping on ARM is a single instruction whereas this byte level manipulation is several more as ARM is based on doing everything with 32-bit types.

Overall, the supported primitives and render stack need serious optimization.  You're not even using precomputed data for rendering circles and arcs are calculated in realtime!  The overabundance of processing power are their saving grace but every extra cycle shortens the battery life (though we both know it's the radio that are the real power drain).

TL;DR: 10MHz is luxurious and should be enough for everyone! </old man yelling at kids> :)

Reply:
In terms of speed - please feel free to share https://quick-bench.com or goldbolt compiler. I didn't input that selected code but I'm pretty convinced that both a lambda with no context capture as well as a constexpr should be pretty much the same on `-O2` in this particular example.

Saying that it wasn't my goal to promote any approach - it can be even a packed union. What I had in mind was: the more readable and tested code - the better.

Reply:
All good.  The latest seems plenty readable.  I'd only bother with assembly level optimization a la godbolt if the design of the render system and MCU selection wasn't already set in stone as real changes could be made.  Rounding the corners of a square wheel may make it roll better but it'll never compare to a circular wheel.